infrastructure/comms/src/connection/peer_connection/connection.rs

```diff
@@ -133,56 +188,282 @@ impl PeerConnection {
     ///
     /// `frames` - The frames to send
     pub fn send(&self, frames: FrameSet) -> Result<()> {
-        let thread_messenger = self.get_thread_messenger()?;
-        thread_messenger.send(ControlMessage::SendMsg(frames))
+        self.send_contol_message(ControlMessage::SendMsg(frames))
     }
 
-    /// Returns true if the PeerConnection is in a connected state, otherwise false
-    pub fn is_connected(&self) -> bool {
-        match self.state {
-            PeerConnectionState::Connected(_) => true,
-            _ => false,
-        }
+    /// Temporarily suspend messages from being processed and forwarded to the consumer.
+    /// Pending messages will be buffered until reaching the receive HWM. Once resumed,
+    /// buffered messages will be released to the consumer.
+    /// An Err will be returned if the connection is not in a Connected state.
+    pub fn pause(&self) -> Result<()> {
+        self.send_contol_message(ControlMessage::Pause)
+    }
+
+    /// Unpause the connection and resume message processing from the peer.
+    /// An Err will be returned if the connection is not in a Connected state.
+    pub fn resume(&self) -> Result<()> {
+        self.send_contol_message(ControlMessage::Resume)
     }
 
-    /// Returns the ThreadMessenger. Will return an Err if the PeerConnection is not in
-    /// a connected state.
-    fn get_thread_messenger(&self) -> Result<&ThreadControlMessenger> {
-        match self.state {
-            PeerConnectionState::Connected(ref state) => Ok(state),
-            _ => Err(PeerConnectionError::StateError(format!(
+    /// Send control message to the ThreadControlMessenger.
+    /// Will return an error if the connection is not in an active state.
+    ///
+    /// # Arguments
+    ///
+    /// `msg` - The ControlMessage to send
+    fn send_contol_message(&self, msg: ControlMessage) -> Result<()> {
+        use PeerConnectionState::*;
```

Comment:
hmm, I agree if your method is huge, but in small methods which do one thing is that really the case? I see this as a quite neat pattern, a function local `use {enum}::*` followed by the match statement which uses the values. Especially if you have multiple matches in one branch. Is this the first time you're seeing this used? If so, does the syntax make sense now that you know about it? I'm open to changing it, but I find it much less verbose and easier to read.

Reply:
I just find it possible to be ambiguous (my IDE agrees). In this case I agree it is fine because there are no other global enum or structs being referenced in this function, but as soon as there is even one struct or enum being referenced from outside that `use` it will be unclear where they come from.

It is just a nit and I agree that in both the cases I brought this up there was no ambiguity but I twitched at the possibility of ambiguity.

Reply:
Ye my IDE also doesn't cover this case either (though it should). I think as long as you import in the function scope and your method is one match only relating to one enum, it's clear and very helpful for bigger matches (see worker.rs).

Reply:
Agreed